packages/algolia-experiences/src/setup-instantsearch.ts

```diff
@@ -27,45 +27,33 @@ export function setupInstantSearch() {
     });
     window.__search = search;
 
-    if (!customElements.get('algolia-experience')) {
-      registerComponents(search, settings);
-    }
-
-    const elements = getElements();
-
     injectStyles();
 
-    fetchConfiguration([...elements.keys()], settings).then((configuration) => {
-      search
-        .addWidgets(
-          configuration.flatMap((config) => configToIndex(config, elements))
-        )
-        .start();
-    });
+    if (!customElements.get('algolia-experience')) {
+      registerComponent(search, settings);
+    }
   } catch (err) {
     error((err as Error).message);
   }
 }
 
-function registerComponents(search: InstantSearch, settings: Settings) {
+function registerComponent(search: InstantSearch, settings: Settings) {
   class AlgoliaExperience extends HTMLElement {
     static observedAttributes = ['experience-id'];
 
     widgets: IndexWidget[] = [];
 
     connectedCallback() {
-      const experienceId = this.getAttribute('experience-id');
-      if (!experienceId) {
+      const id = this.getAttribute('experience-id');
```

Comment:
would there be less repetition if we call the attribute `data-id`? can swap before the GA still